### ISSUE TITLE
Option to enable/disable notification (and search parameter)

### ISSUE THREAD

**Issue by rightaway** [mchristofides/pg_docs_bot#10]
Great addon! Could the addon disable the URL bar query change (`pg-docs-bot-redirected`) and the notification in the bottom right (`Redirected by pg_docs_bot from`)? Instead like many addons do, the icon could be grayed out at all times unless it was redirected, then the icon can have colors. And if you click on the icon it'll show you the link to the page it was redirected from. I think it would be less intrusive and more useful that way.

**Comment by mchristofides** [comment]
Thanks! I'm afraid the search parameter in the URL is currently how it communicates the redirected URL. The icon is a neat idea, I hadn't considered that! However, I like that I don't need another extension icon visible, so I wonder if making the notice less intrusive visually could help instead? Perhaps a single line with the URL linked?

**Comment by rightaway** [comment]
The notification is useful when someone wants to go to the version the search engine took them to, which would only happen by luck. People either want to see docs for the current version or the version they'll set in https://github.com/mchristofides/pg_docs_bot/issues/1. Maybe the notification isn't necessary?

**Comment by mchristofides** [comment]
Yes good point, and the search engine use case is definitely the main one!

A few reasons I currently include it:
* In case somebody searched an exact version number ([example from the mailing lists](https://www.postgresql.org/message-id/B884788A-B65C-4E5F-8495-0F01F166FD39%40thebuild.com))
* In case a StackOverflow question / answer was about a specific version
* In case the page 404s (eg a deprecated feature that pg_docs_bot doens't know about)

For starters, to make it less intrusive, I've made the notice a single line by hyperlinking just the version number, and added a close button. I'll keep thinking about the need for it at all, and also consider alternatives to the search parameter implementation!

What do you think?

![single-line-version-number-close](https://user-images.githubusercontent.com/10532911/80865862-cdd34180-8c83-11ea-9f11-f12eadfb3a27.png)

**Comment by rightaway** [comment]
The smaller notification is nicer. An option to enable/disable it would be good if you add settings to this addon.

**Comment by mchristofides** [comment]
Brilliant, good idea, thank you. I'll change the name of this issue to remind me!

**Comment by rightaway** [comment]
Why does the updating version require new permissions, like accessing data on google.com and bing.com domains? Regardless of what source page the link is on, as long as you have access to data on postgresql.org shouldn't it be enough to perform a redirect?

**Comment by mchristofides** [comment]
@rightaway sorry I haven't done anything on this since May, so it took a bit of dusting off!

I went with an implementation that uses `chrome.webRequest.onBeforeRequest`, to make it easier to ignore things like viewing an older version when you're actually on the docs, and to avoid 404s on certain deprecated features. Sadly this approach meant needing permissions to the source pages. I think I originally asked for <all_urls> permissions in pre-1.0 versions, which seemed way too much, so I reduced down to the list you'll have seen (common search engines and Stack Exchange sites). Hope that makes sense.

It looks like somebody has made one that works more like you're suggesting, in case that would be your preferred approach?
https://chrome.google.com/webstore/detail/postgres-docs-redirect/ajcfpdiobeccopppdonocgddplegokbl

**Comment by mchristofides** [comment]
@rightaway just heard about a much neater solution on the way, which seems to need fewer permissions, stores the last version viewed, and also doesn't mess with the URL. Looks promising! https://github.com/mchristofides/pg_docs_bot/issues/1#issuecomment-754944226
